src/server/utils/cache.ts

```
@@ -0,0 +1,116 @@
+
+function isArrayLike(obj: any): boolean {
+  // Check if it's an object but not null
+  if (!obj || typeof obj !== 'object' || Array.isArray(obj)) {
+    return false;
+  }
+
+  // Check if all keys are numeric indices
+  const keys = Object.keys(obj);
+  return keys.every((key) => {
+    const parsed = parseInt(key, 10);
+    return !isNaN(parsed) && parsed >= 0 && parsed.toString() === key;
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)